src/main/flight/gps_rescue.c

```diff
@@ -395,25 +403,53 @@ void rescueAttainPosition()
 }
 
 // Very similar to maghold function on betaflight/cleanflight
-void setBearing(int16_t deg)
+void setBearing(int16_t desiredHeading)
 {
-    int16_t dif = DECIDEGREES_TO_DEGREES(attitude.values.yaw) - deg;
+    float errorAngle = DECIDEGREES_TO_DEGREES(attitude.values.yaw) - desiredHeading;
+
+    // Determine the most efficient direction to rotate
+    if (errorAngle <= -180) {
+        errorAngle += 360;
+    } else if (errorAngle > 180) {
+        errorAngle -= 360;
+    }
 
-    if (dif <= -180) {
-        dif += 360;
-    } else if (dif > 180) {
-        dif -= 360;
+    errorAngle *= -GET_DIRECTION(rcControlsConfig()->yaw_control_reversed);
+    const int errorAngleSign = (errorAngle < 0) ? -1 : 1;
+    const float errorAngleAbs = ABS(errorAngle);
+        
+    // Calculate a desired yaw rate based on a maximum limit beyond
+    // an error window and then scale the requested rate down inside
+    // the window as error approaches 0.
+    if (errorAngleAbs > GPS_RESCUE_RATE_SCALE_DEGREES) {
+        rescueYaw = GPS_RESCUE_MAX_YAW_RATE;
+    } else {
+        rescueYaw = (errorAngleAbs / GPS_RESCUE_RATE_SCALE_DEGREES) * GPS_RESCUE_MAX_YAW_RATE;
     }
+    rescueYaw = (errorAngleAbs > GPS_RESCUE_RATE_SCALE_DEGREES) ? GPS_RESCUE_MAX_YAW_RATE : errorAngleAbs;
 
-    dif *= -GET_DIRECTION(rcControlsConfig()->yaw_control_reversed);
+    rescueYaw *= errorAngleSign;
+}
 
-    rescueYaw = - (dif * gpsRescueConfig()->yawP / 20);
+float gpsRescueGetYawRate(void)
+{
+    return rescueYaw;
 }
 
-void gpsRescueInjectRcCommands(void)
+float gpsRescueGetThrottle(void)
 {
-    rcCommand[THROTTLE] = rescueThrottle;
-    rcCommand[YAW] = rescueYaw;
+    float commandedThrottle = 0;
+    const int minCheck = MAX(rxConfig()->mincheck, 1000);
+    
+    // We need to compensate for min_check since the throttle value set by gps rescue
+    // is based on the raw rcCommand value commanded by the pilot.
+    if (rescueThrottle > minCheck) {
+        // Calculated a desired commanded throttle scaled from 0.0 to 1.0 for use in the mixer.
+        commandedThrottle = (float)(rescueThrottle - minCheck) / (PWM_RANGE_MAX - minCheck);
```

Comment:
```
fScaleRange(rescueThrottle, MAX(rxConfig()->mincheck, 1000), PWM_RANGE_MAX, 0.0f, 1.0f);
commandedThrottle = constrainf(commandedThrottle, 0.0f, 1.0f);
```
BTW: probably PWM_RANGE_MIN instead of 1000

Reply:
We don't have a float version of `scaleRange`.

Reply:
It can be part of this PR or I can prepare separate one.
IMO scaleRange is much easier to read ...